web/packages/teleport/src/Discover/SelectResource/SelectResource.tsx

```diff
@@ -146,6 +191,43 @@ export function SelectResource({ onSelect }: SelectResourceProps) {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  async function updatePinnedGuides(guideId: DiscoverGuideId) {
+    const guidePreferences = preferences.discoverResourcePreferences;
```

Comment:
why change the name? whats wrong with this?
```suggestion
    const { discoverResourcePreferences } = preferences;
```